wpa_supplicant/ctrl_iface_zephyr.c

```diff
@@ -8,6 +8,124 @@
 
 #include "ctrl_iface_zephyr.h"
 
+
+static int wpa_supplicant_ctrl_mon_iface_attach(struct wpa_ctrl_mon **head, int sock)
+{
+	struct wpa_ctrl_mon *dst;
+
+	dst = os_zalloc(sizeof(*dst));
+	if (dst == NULL)
+		return -1;
+
+	dst->sock = sock;
+	dst->debug_level = MSG_INFO;
+	dst->next = *head;
+	*head = dst;
+	return 0;
+}
+
+
+static int wpa_supplicant_ctrl_mon_iface_detach(struct wpa_ctrl_mon **head, int sock)
+{
+	struct wpa_ctrl_mon *dst, *prev = NULL;
+
+	dst = *head;
+	while (dst) {
+		if (dst->sock == sock) {
+			if (prev == NULL) {
+				*head = dst->next;
+			} else {
+				prev->next = dst->next;
+			}
+			os_free(dst);
+			return 0;
+		}
+		prev = dst;
+		dst = dst->next;
+	}
+	return -1;
+}
+
+static void wpa_supplicant_ctrl_iface_send(struct wpa_supplicant *wpa_s,
+					   const char *ifname, int sock,
+					   struct wpa_ctrl_mon **head,
+					   int level, const char *buf,
+					   size_t len)
+{
+	struct wpa_ctrl_mon *dst, *next;
+	char levelstr[64];
+	int idx;
+	char *sbuf;
+	struct conn_msg msg;
+
+	dst = *head;
+	if (sock < 0 || dst == NULL)
+		return;
+
+	if (ifname)
+		os_snprintf(levelstr, sizeof(levelstr), "IFNAME=%s <%d>",
+			    ifname, level);
+	else
+		os_snprintf(levelstr, sizeof(levelstr), "<%d>", level);
+
+	sbuf = os_malloc(len);
```

Comment:
Why is this intermediate buffer needed? Can't we just use `conn_msg.msg` itself?

Reply:
Done